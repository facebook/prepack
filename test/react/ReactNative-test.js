/**
 * Copyright (c) 2017-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

/* @flow */

const setupReactTests = require("./setupReactTests");
const { runTest } = setupReactTests();

// $FlowFixMe: Jest is already defined globally
jest.unmock("../../node_modules/react-native/Libraries/Components/View/View.js");
// $FlowFixMe: Jest is already defined globally
jest.unmock("../../node_modules/react-native/Libraries/Text/Text.js");

/* eslint-disable no-undef */
const { it } = global;

it("Simple", async () => {
  runTest(__dirname + "/ReactNative/simple.js");
});

it("Simple 2", async () => {
  runTest(__dirname + "/ReactNative/simple2.js");
});
