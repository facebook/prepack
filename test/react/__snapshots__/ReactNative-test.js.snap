// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple 2: (JSX => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 9,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": Array [],
                              "message": "",
                              "name": "Context.Consumer",
                              "status": "INLINED",
                            },
                          ],
                          "message": "",
                          "name": "TouchableText",
                          "status": "INLINED",
                        },
                      ],
                      "message": "",
                      "name": "",
                      "status": "FORWARD_REF",
                    },
                  ],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 8,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple 2: (JSX => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 9,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": Array [],
                              "message": "",
                              "name": "Context.Consumer",
                              "status": "INLINED",
                            },
                          ],
                          "message": "",
                          "name": "TouchableText",
                          "status": "INLINED",
                        },
                      ],
                      "message": "",
                      "name": "",
                      "status": "FORWARD_REF",
                    },
                  ],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 8,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple 2: (createElement => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 9,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": Array [],
                              "message": "",
                              "name": "Context.Consumer",
                              "status": "INLINED",
                            },
                          ],
                          "message": "",
                          "name": "TouchableText",
                          "status": "INLINED",
                        },
                      ],
                      "message": "",
                      "name": "",
                      "status": "FORWARD_REF",
                    },
                  ],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 8,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple 2: (createElement => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 9,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": Array [],
                              "message": "",
                              "name": "Context.Consumer",
                              "status": "INLINED",
                            },
                          ],
                          "message": "",
                          "name": "TouchableText",
                          "status": "INLINED",
                        },
                      ],
                      "message": "",
                      "name": "",
                      "status": "FORWARD_REF",
                    },
                  ],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 8,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple: (JSX => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 7,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 6,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple: (JSX => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 7,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 6,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple: (createElement => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 7,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 6,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple: (createElement => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 7,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Consumer",
                  "status": "INLINED",
                },
                Object {
                  "children": Array [],
                  "message": "",
                  "name": "Context.Provider",
                  "status": "INLINED",
                },
              ],
              "message": "",
              "name": "TouchableText",
              "status": "INLINED",
            },
          ],
          "message": "",
          "name": "",
          "status": "FORWARD_REF",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 6,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;
